build.yaml

```diff
@@ -0,0 +1,575 @@
+# vim ft=yaml:fdm=indent
+# To modify an existing job, simply update the dictionary 'jobs_type'
+# commit, push
+# Then click 'deploy' on the jenkins job related to this repository
+# available on the jenkins homepage.
+
+project: 'ClusterHQ/flocker'
+git_url: 'https://github.com/ClusterHQ/flocker.git'
+
+# common_cli contains YAML aliases for common cli operations
+# used during the build process
+common_cli:
+  hashbang: &hashbang |
+    #!/bin/bash -l
+    set -x
+    set -e
+    os() {
+      # Determine OS platform
+      test -e /home/centos && DISTRO="CentOS"
+      test -e /home/ubuntu && DISTRO="Ubuntu"
+      echo \$DISTRO
+    }
+    # add a function that we can consume to colorise output
+    # https://wiki.archlinux.org/index.php/Color_Bash_Prompt
+    parse_logs() {
+      # ubuntu defaults to mawk, which causes some things to break
+      case "\$(os)"
+      in
+        "Xubuntu") alias awk='mawk -W interactive'
+      esac
+
+      "\$@" | awk '
+        # exceptions to rules below, these are always printed black
+        /.*reading.sources.*error_pages.404.*/ {print "\\033[0;30m" \$0 "\\033[0;30m";  next }
+        /.*writing.output.*error_pages.404.*/ {print "\\033[0;30m" \$0 "\\033[0;30m";  next }
+        /.*errors.py.*/ {print "\\033[0;30m" \$0 "\\033[0;30m";  next }
+
+        # red lines
+        /.*exception.*/ {print "\\033[0;31m" \$0 "\\033[0;30m";  next }
+        /.*Exception.*/ {print "\\033[0;31m" \$0 "\\033[0;30m";  next }
+        /.*fail.*/      {print "\\033[0;31m" \$0 "\\033[0;30m";  next }
+        /.* ERROR.*/     {print "\\033[0;31m" \$0 "\\033[0;30m";  next }
+        /.* error.*/     {print "\\033[0;31m" \$0 "\\033[0;30m";  next }
+        /build finished with problems./     {print "\\033[0;31m" \$0 "\\033[0;30m";  next }
+        /return 1/     {print "\\033[0;31m" \$0 "\\033[0;30m";  next }
+        /Could not find valid repo at/     {print "\\033[0;31m" \$0 "\\033[0;30m";  next }
+
+        # yellow lines
+        /.*Warning.*/   {print "\\033[0;33m" \$0 "\\033[0;30m";  next }
+        /.*warning.*/   {print "\\033[0;33m" \$0 "\\033[0;30m";  next }
+        /.*skip.*/      {print "\\033[0;33m" \$0 "\\033[0;30m";  next }
+
+        # green lines
+        /.*succeeded.*/  {print "\\033[0;32m" \$0 "\\033[0;30m";  next }
+        /.*Succeeded.*/  {print "\\033[0;32m" \$0 "\\033[0;30m";  next }
+        /PASSED.*/  {print "\\033[0;32m" \$0 "\\033[0;30m";  next }
+        /.*[OK]/  {print "\\033[0;32m" \$0 "\\033[0;30m";  next }
+
+        # everything else, print in black
+        /.*/            {print "\\033[0;30m" \$0 "\\033[0;30m" ; }
+      '
+      return \${PIPESTATUS[0]}
+    }
+    # fix the docker permission issue
```

Comment:
related to what? the colors used ? 

    1   | # add a function that we can consume to colorise output                                                                                                                                             
  23      # https://wiki.archlinux.org/index.php/Color_Bash_Prompt